qiskit/circuit/quantumcircuit.py

```diff
@@ -1165,6 +1166,20 @@ def append(
             if is_parameter:
                 instruction = copy.deepcopy(instruction)
 
+        # Deprecate the passing multiple qargs for contolled gates
+        if len(qargs) == 2:  # check if singly-controlled gate
+            # check for multiple qubits
+            if any(isinstance(qarg, list) and len(qarg) > 1 for qarg in qargs):
+                warnings.warn(
+                    "Creating multiple single-controlled gates with a single call, "
+                    "such as QuantumCircuit.cx([0, 1], 2) is deprecated as of Qiskit "
+                    "Terra 0.19.0 and will be removed no sooner than 3 months after it's "
+                    "release date. To construct multiple gates, call the according "
+                    "method multiple times with a single control and target.",
+                    DeprecationWarning,
+                    stacklevel=3,
+                )  # stacklevel 3 to show the correct method call
+
```

Comment:
If we are keeping this broadcasting behaviour for `rxx` and friends (I don't remember what was decided in the meeting), then it might be worth moving the warnings into the actual `QuantumCircuit.cx` gate calls, and so on.  When we add the new behaviour, the tests for multiple inputs will have to move there anyway.

That _would_ mean that the user would still be able to achieve broadcasting by using `QuantumCircuit.append` directly, but that's not necessarily a bad thing; we've generally been treating `QuantumCircuit.append` as the advanced use-case, which you can use to get more powerful behaviour in certain cases (adding custom gates, etc), whereas the `cx`, `x`, whatever, methods are convenient shorthands.

Having come back to look at this PR again, that'd probably be my preferred way of doing things, but I don't know about others.

Reply:
I agree with this, I tried to avoid it since it means touching a lot of functions but so be it 🤷🏻